odoo/models.py

```diff
@@ -1625,34 +1625,33 @@ def load_views(self, views, options=None):
         options = options or {}
         result = {}
 
-        toolbar = options.get('toolbar')
-        result['fields_views'] = {
-            v_type: self.fields_view_get(v_id, v_type if v_type != 'list' else 'tree',
-                                         toolbar=toolbar if v_type != 'search' else False)
+        result['views'] = {
+            v_type: self.get_view(
+                v_id, v_type if v_type != 'list' else 'tree',
+                **options
+            )
             for [v_id, v_type] in views
         }
-        result['fields'] = self.fields_get()
-
-        if options.get('load_filters'):
-            result['filters'] = self.env['ir.filters'].get_filters(self._name, options.get('action_id'))
+        models = set(model for view_type, info in result['views'].items() for model in info.pop('models'))
```

Comment:
`view_type` is unused
```suggestion
        models = {model for info in result['views'].values() for model in info.pop('models')}
```